src/scaffold/tests.js

```diff
@@ -0,0 +1,185 @@
+    return `let ${input.name || `param${index}`} = ${assignValue(ascType, input.name)}`
+  }).join('\n')
+}
+
+// Generates the value that will be assigned to a variable in generateArguments()
+const assignValue = (type) => {
+  switch (type) {
+    case "string":
+      return `"${VARIABLES_VALUES[type]}"`
+    case "BigInt":
+      return `BigInt.fromI32(${VARIABLES_VALUES[type]})`
+    case "Address":
+      return `Address.fromString("${VARIABLES_VALUES[type]}")`
+    case "Bytes":
+      return `Bytes.fromI32(${VARIABLES_VALUES[type]})`
+    case type.match(/Array<(.*?)>/)?.input:
+      innerType = type.match(/Array<(.*?)>/)[1]
+      return `[${assignValue(innerType)}]`
+    default:
+      let value = VARIABLES_VALUES[type]
+      return value ? value : `"${type} Not implemented"`
+  }
+}
+
+/*
+  Generates the assert.fieldEquals() for a given entity and event inputs. Example:
+  assert.fieldEquals(
+    "ExampleEntity",
+    "0xa16081f360e3847006db660bae1c6d1b2e17ec2a",
+    "owner",
+    "0x0000000000000000000000000000000000000001"
+  )
+*/
+const generateFieldsAssertions = (entity, eventInputs, indexEvents) => eventInputs.filter(input => input.name != "id").map((input, index) =>
+  `assert.fieldEquals(
+    "${entity}",
+    "0xa16081f360e3847006db660bae1c6d1b2e17ec2a${indexEvents ? "-1" : ""}",
+    "${input.name || `param${index}`}",
+    "${expectedValue(ascTypeForEthereum(input.type))}"
+  )`
+).join('\n')
+
+// Returns the expected value for a given type in generateFieldsAssertions()
+const expectedValue = type => {
+  switch (type) {
+    case type.match(/Array<(.*?)>/)?.input:
+      innerType = type.match(/Array<(.*?)>/)[1]
+      return `[${expectedValue(innerType)}]`
+    default:
+      let value = VARIABLES_VALUES[type]
+      return value ? value : `${type} Not implemented`
+  }
+}
+
+// Checks if the type is a native AS type or should be imported from graph-ts
+const isNativeType = type => {
+  let natives = [
+    /Array<([a-zA-Z0-9]+)?>/,
+    /i32/,
+    /string/,
+    /boolean/
+  ]
+
+  return natives.some(rx => rx.test(type));
+}
+
+// Generates the example test.ts file
+const generateExampleTest = (contract, event, indexEvents, importTypes) => {
+  const entity = indexEvents ? `${event._alias}` : 'ExampleEntity'
+  const eventInputs = event.inputs
+  const eventName = event._alias
+
+  return `
+  import { assert, describe, test, clearStore, beforeAll, afterAll} from "matchstick-as/assembly/index"
+  import { ${importTypes} } from "@graphprotocol/graph-ts"
+  import { ${entity} } from "../generated/schema"
+  import { ${indexEvents ? `${eventName} as ${eventName}Event` : eventName} } from "../generated/${contract}/${contract}"
+  import { handle${eventName} } from "../src/${strings.kebabCase(contract)}"
+  import { create${eventName}Event } from "./${strings.kebabCase(contract)}-utils"
+
+
+  // Tests structure (matchstick-as >=0.5.0)
+  // https://thegraph.com/docs/en/developer/matchstick/#tests-structure-0-5-0
```

Comment:
I thought about maybe in the future `matchstick` providing the test file as a template (eg: `handlebars`), but maybe it's better to generate it fully here in the `graph-cli`. Just wondering really, for now I think the way it is at the moment is great 🙂 👍

Reply:
We thought that it is more fitting to add it to graph-cli, because it handles generation of other similar files and already has the helpers needed (the conversion functions provided by `src/codegen/types` were a massive help)